src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,132 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String TITLE = "title";
+    private static final String PRICE = "price";
+    private static final String ID = "id";
+    private static final int MIN_AFFECTED_ROWS = 1;
+
+    @Override
+    public Book create(Book book) {
+        String insertQuery = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(insertQuery,
+                        Statement.RETURN_GENERATED_KEYS)) {
+
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            checkRows(affectedRows, "Expected to insert at least 1 row, but inserted 0 rows.");
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(MIN_AFFECTED_ROWS, Long.class);
```

Comment:
this constant usage only confuses me more, we can not reuse the same constant in 2 different places just because it has the same value, it should add context and understanding with it's name
```suggestion
                Long id = generatedKeys.getObject(GENERATED_KEY_COLUMN_INDEX, Long.class);
```
or use column name instead, I think it is possible